### PR TITLE
Fix some clang-10 warnings

### PR DESCRIPTION
**Pull request by masterleinad**
Among others, also fixes the warnings in https://cdash.43-1.org/viewBuildError.php?type=1&buildid=5988.

### CODE REVIEW

File: bundled/umfpack/UMFPACK/Source/umf_grow_front.cc [hunk 1]
```diff
@@ -112,7 +112,7 @@ GLOBAL Int UMF_grow_front
 	/* :: frontal matrix size int overflow :: */
 	/* the desired front size is bigger than the integer maximum */
 	/* compute a such that a*a*s < Int_MAX / sizeof (Entry) */
-	double a = 0.9 * sqrt ((Int_MAX / sizeof (Entry)) / s) ;
+	double a = 0.9 * sqrt (((double) Int_MAX / sizeof (Entry)) / s) ;
```
Comment (peterrum) [inline bundled/umfpack/UMFPACK/Source/umf_grow_front.cc:115]: Is it intended that you have changed also UMFPACK?

Comment (masterleinad) [reply bundled/umfpack/UMFPACK/Source/umf_grow_front.cc:115]: Yes, that is an extra commit. The warning is along the lines that the implicit cast to `double` changes the value of `Int_MAX`.

Comment (rezarastak) [inline bundled/umfpack/UMFPACK/Source/umf_grow_front.cc:115]:
```suggestion
	double a = 0.9 * sqrt ((double(Int_MAX) / sizeof (Entry)) / s) ;
```
Doesn't clang-tidy warn also about c-style casts?

Comment (tjhei) [reply bundled/umfpack/UMFPACK/Source/umf_grow_front.cc:115]: This is clang-10, not clang-tidy-10, but I assume that we disable c-style cast warnings in umfpack.

Comment (masterleinad) [reply bundled/umfpack/UMFPACK/Source/umf_grow_front.cc:115]: This is also the style used in the rest of the file and I decided to rather follow it than to go with what we would do in the (main) library.

**Pull request merged**